docs/reference/src/main/asciidoc/ri-spi.asciidoc

```diff
@@ -979,3 +979,65 @@ Instead, this dependency needs to be deployed separately to the OSGi container.
 ** Deprecated `org.jboss.weld.environment.se.ShutdownManager` was removed.
 
 ** Deprecated `org.jboss.weld.environment.se.contexts.interceptors.ActivateThreadScope` was removed.
+
+==== Migration from Weld 2.x to 3.0
+
+Weld 3.0 implements CDI 2.0 so this is a huge update which contains bunch of new features.
+However, this document only covers part which might prove problematic while moving existing application from Weld 2.x to Weld 3.0.
+For more information on the CDI 2.0 itself, please consult http://docs.jboss.org/cdi/spec/2.0-PRD/cdi-spec.html[the specification]
+
+===== Java 8
+
+Both, Weld 3.0 and CDI 2.0 require Java 8 for compile time and runtime.
+
+===== Weld-core artifact change 
+
+For historical reasons, Weld 2.x core is released in two versions which *are identical* in their contents.
+Those are `org.jboss.weld:weld-core` and `org.jboss.weld:weld-core-impl`.
+Starting with Weld 3.0, only `org.jboss.weld:weld-core-impl` artifact will be released.
+
+===== Shaded JARs name change
+
+Weld 2.x created shaded JARs for SE and Servlet with following artifact ID: `weld-se`, `weld-servlet`.
+In Weld 3.0 a `-shaded` affix was added to the name of these artifacts to make it clear what they contain.
+E.g. new artifact IDs of these JARs are `weld-se-shaded` and `weld-servlet-shaded` respectively.
+
+===== Weld 3 modules
+
+Weld 3.0 was fine-grained and several modules were separated from the core.
```

Comment:
I would use "optional modules" so that it's clear that those modules are only required if a given technology is present/integrated.